src/sgame/sg_main.cpp

```diff
@@ -37,7 +37,7 @@ level_locals_t level;
 gentity_t          *g_entities;
 gclient_t          *g_clients;
 
-Cvar::Range<Cvar::Cvar<int>> g_showHelpOnConnection("g_showHelpOnConnection", "show MN_WELCOME on connect (1 = all, 2 = unregistered)", Cvar::NONE, 1, 0, 2);
+Cvar::Range<Cvar::Cvar<int>> g_showHelpOnConnection("g_showHelpOnConnection", "show MN_WELCOME on connect (1 = all, 2 = unregistered)", Cvar::NONE, 0, 0, 2);
```

Comment:
We should rename this from g_showHelpOnConnection to something more reasonable like g_showWelcome.

We can track state like whether someone has seen it (track a generation or something) in a separate location (admin.dat or something...)

Reply:
This seems overall out of the scope of this PR, which only tries to repair a feature that existed.

> We should rename this from g_showHelpOnConnection to something more reasonable like g_showWelcome.

I usually try to avoid changing an interface (especially when it's something people can have tweaked, and won't have any way to know they're using a cvar that no longer exists!), even if I do agree with you a lot on the name there.

> We can track state like whether someone has seen it (track a generation or something) in a separate location (admin.dat or something...)

feels out of scope to me.